### PR TITLE
[DOC] Add set processor documentation

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Closes content gap

### Issues Resolved
#4647 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@hdhalter We need a dev to take the first pass at drafting this content. I can support them in refining the content and getting it through the documentation process. The ingest processor template is provided with this PR, so it should help the dev get started. Thanks!

File: _ingest-pipelines/processors/set.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+<insert syntax example>
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `set` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
```
Comment (vagimeli) [inline _ingest-pipelines/processors/set.md:24]: @gaobinlong Please confirm all parameters are correct and listed for this processor.

**Comment by vagimeli** [comment]
@gaobinlong This PR is ready for your review at your availability. Thank you for all your technical reviews on these topics!

File: _ingest-pipelines/processors/set.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+---
+layout: default
+title: Set
+parent: Ingest processors
+nav_order: 240
+---
+
+# Set processor
+
+The `set` processor adds or updates fields in a document. It sets one field and associates it with the specified value. If the field already exists, its value is replaced with the provided one.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/set.md:10]:
```suggestion
The `set` processor adds or updates fields in a document. It sets one field and associates it with the specified value. If the field already exists, its value is replaced with the provided one, unless the override parameter is set to false. When override is false and the specified field exists, the value of the field will remain unchanged.
```

Comment (vagimeli) [reply _ingest-pipelines/processors/set.md:10]: @gaobinlong Please confirm the change accurately reflects your comment. Thank you.

File: _ingest-pipelines/processors/set.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `set` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The name of the field to be set or updated. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`value` | Required | The value to be assigned to the field. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`overrideEnabled` | Optional | A Boolean flag that determines whether the processor should override the exisitng value of the field.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/set.md:37]:
```suggestion
`override` | Optional | A Boolean flag that determines whether the processor should override the existing value of the field.
```

File: _ingest-pipelines/processors/set.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+`ignoreEmptyValue` | Optional | A Boolean flag that determines whether the processor should ignore empty values.
```
Comment (vagimeli) [inline _ingest-pipelines/processors/set.md:38]:
```suggestion
`ignore_empty_value` | Optional | A Boolean flag that determines whether the processor should ignore `null` values or empty strings. Default is `false`.
```

File: _ingest-pipelines/processors/set.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+POST testindex1/_doc
```
Comment (vagimeli) [inline _ingest-pipelines/processors/set.md:120]:
```suggestion
POST testindex1/_doc?pipeline=set-pipeline
```

File: _ingest-pipelines/processors/set.md (hunk 2 above)
Comment (natebower) [reply _ingest-pipelines/processors/set.md:10]: Edited suggestion

File: _ingest-pipelines/processors/set.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The name of the field to be set or updated. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`value` | Required | The value to be assigned to the field. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets).
+`override` | Optional | A Boolean flag that determines whether the processor should override the existing value of the field.
+`ignore_empty_value` | Optional | A Boolean flag that determines whether the processor should ignore `null` values or empty strings. Default is `false`.
+`description`  | Optional  | A description of the processor's purpose or configuration.
+`if` | Optional | Specifies to conditionally execute the processor.
+`ignore_failure` | Optional | Specifies to ignore failures for the processor. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
```
Comment (natebower) [inline _ingest-pipelines/processors/set.md:41]:
```suggestion
`ignore_failure` | Optional | Specifies to ignore processor failures. See [Handling pipeline failures]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
```

**Comment by vagimeli** [comment]
@natebower Thank you for the thorough review. I've accepted your edits. Will you approve for merging? This is part of the series that I opened PRs and co-authored with dev, but can't approve. Thank you.

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**